go-controller/pkg/ovn/egressgw_test.go

```diff
@@ -2052,7 +2052,70 @@ var _ = ginkgo.Describe("OVN Egress Gateway Operations", func() {
 			err := app.Run([]string{app.Name})
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
+		ginkgo.It("delete legacy hybrid route policies", func() {
```

Comment:
Given that you've added a test, could you also add one for non-legacy LRPs (using address set) and verify that they don't get deleted?

Reply:
can be done in this test itself I think...

Reply:
done